### DISCUSSION TITLE
Scaling Pimcore with multiple instances

### DISCUSSION THREAD

**Discussion by vriveramgmglobant** [pimcore/pimcore#15800] [Q&A / Support] [answered]
Hi! In my current project, we are implementing a load balancer and scaling to multiple instances depending on the server's demand, the problem is we are getting the "Detected CSRF Attack! Do not do evil things with pimcore … ;-)" error. It makes sense because the csrf token is being generated in different instances hence it's not matching when validating it. Is there a workaround or any suggestions? 

I saw this in the documentation, so I'm wondering if it's a bad idea to find an alternative solution for my issue.

```
Cluster Setup
To be done ...

Filesystem
To be done ...

Do not use rsync or other nasty solutions to share the data between the application servers.
```

Thanks in advance

**Comment by NiklasBr** [comment]
I think you can use [Sticky Sessions](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/sticky-sessions.html) to resolve this. ([Azure link](https://learn.microsoft.com/en-us/azure/container-apps/sticky-sessions?pivots=azure-portal))

**Reply by dpfaffenbauer** [reply, marked as answer]
Never use Sticky Sessions, that destroys the complete use-case of horizontal scaling. The CSRF Token is stored in Session and therefore your session has to be stored in a central storage like DB or Redis.

**Comment by vriveramgmglobant** [comment]
> Never use Sticky Sessions, that destroys the complete use-case of horizontal scaling. The CSRF Token is stored in Session and therefore your session has to be stored in a central storage like DB or Redis.

Thanks for your suggestion, I'm gonna try it.

**Comment by vriveramgmglobant** [comment]
It worked with setting up Redis as central storage. Thanks, @dpfaffenbauer

**Reply by dpfaffenbauer** [reply]
perfect.
